tests/import_export_tests.py

```diff
@@ -0,0 +1,386 @@
+        self.assertEquals(
+            {'remote_id': 1002, 'import_time': 1990, 'database_name': 'main'},
+            json.loads(imported_table_2.json_metadata))
+
+        # Case 3. Import table with 2 metrics and 2 columns
+        table_3 = create_table(
+            'table_3', id=1003, cols_names=['col1', 'col2'],
+            metric_names=['metric1', 'metric2'])
+        imported_table_id_3 = models.SqlaTable.import_obj(
+            table_3, import_time=1991)
+
+        imported_table_3 = get_table(imported_table_id_3)
+        self.assertEquals('table_3', imported_table_3.table_name)
+        self.assertEquals(2, len(imported_table_3.columns))
+        self.assertEquals(
+            {'col1', 'col2'},
+            set([c.column_name for c in imported_table_3.columns]))
+        self.assertEquals(2, len(imported_table_3.metrics))
+        self.assertEquals(
+            {'metric1', 'metric2'},
+            set([m.metric_name for m in imported_table_3.metrics]))
+        imported_table_3_id = imported_table_3.id
+
+        # Case 4. Override table with different metrics and columns
+        table_4 = create_table(
+            'table_3', id=1003, cols_names=['new_col1', 'col2', 'col3'],
+            metric_names=['new_metric1'])
+        imported_table_id_4 = models.SqlaTable.import_obj(
+            table_4, import_time=1992)
+
+        imported_table_4 = get_table(imported_table_id_4)
+        self.assertEquals('table_3', imported_table_4.table_name)
+        self.assertEquals(4, len(imported_table_4.columns))
+        # metrics and columns are never deleted, only appended
+        self.assertEquals(
+            {'col1', 'new_col1', 'col2', 'col3'},
+            set([c.column_name for c in imported_table_4.columns]))
+        self.assertEquals(3, len(imported_table_4.metrics))
+        self.assertEquals(
+            {'new_metric1', 'metric1', 'metric2'},
+            set([m.metric_name for m in imported_table_4.metrics]))
+        # override doesn't change the id
+        self.assertEquals(imported_table_3_id, imported_table_4.id)
+
+        # Case 5. Override with the identical table
+        table_5 = create_table(
+            'table_3', id=1003, cols_names=['new_col1', 'col2', 'col3'],
+            metric_names=['new_metric1'])
+        imported_table_id_5 = models.SqlaTable.import_obj(
+            table_5, import_time=1993)
+
+        imported_table_5 = get_table(imported_table_id_5)
+        self.assertEquals('table_3', imported_table_5.table_name)
+        self.assertEquals(4, len(imported_table_5.columns))
+        self.assertEquals(
+            {'col1', 'new_col1', 'col2', 'col3'},
+            set([c.column_name for c in imported_table_5.columns]))
+        self.assertEquals(3, len(imported_table_5.metrics))
+        self.assertEquals(
+            {'new_metric1', 'metric1', 'metric2'},
+            set([m.metric_name for m in imported_table_5.metrics]))
+
+        # cleanup
```

Comment:
Would if make sense to cleanup at the beginning to setup to test? if the test fails half way, it's nice to be able to assume that you can just re-run the test itself (not the full suite) to fix the state. Idempotent, atomic tests ftw!